devito/symbolics/extended_sympy.py

```diff
@@ -618,6 +652,84 @@ def __str__(self):
     __reduce_ex__ = Pickable.__reduce_ex__
 
 
+class Namespace(sympy.Expr, Pickable):
+
+    """
+    Symbolic representation of a C++ namespace `ns0::ns1::...`.
+    """
+
+    __rargs__ = ('items',)
+
+    def __new__(cls, items, **kwargs):
+        normalized_items = []
+        for i in as_tuple(items):
+            if isinstance(i, str):
+                normalized_items.append(ReservedWord(i))
+            elif isinstance(i, ReservedWord):
+                normalized_items.append(i)
+            else:
+                raise ValueError("`items` must be iterable of str or ReservedWord")
+
+        obj = sympy.Expr.__new__(cls)
+        obj._items = tuple(items)
+
+        return obj
+
+    def _hashable_content(self):
+        return super()._hashable_content() + self.items
+
+    @property
+    def items(self):
+        return self._items
+
+    def __str__(self):
+        return "::".join(str(i) for i in self.items)
+
+    __repr__ = __str__
+
+
+class Rvalue(sympy.Expr, Pickable):
+
+    """
+    A generic C++ rvalue, that is a value that occupies a temporary location in
+    memory.
+    """
+
+    __rargs__ = ('expr',)
+    __rkwargs__ = ('namespace', 'init')
+
+    def __new__(cls, expr, namespace=None, init=None):
+        args = [expr]
```

Comment:
self._expr = expr
self._namespace = namespace
self._init = init

and then pkain properties

Reply:
OMG, not sure what I was thinking when I wrote that garbage. Fixing